packages/@lwc/ssr-compiler/src/compile-template/transformers/component.ts

```diff
@@ -79,7 +79,7 @@ const bYieldFromChildGenerator = esTemplateWithYield`
 `<EsBlockStatement>;
 
 const bAddContent = esTemplate`
-    addContent(${/* slot name */ is.expression} ?? "", async function* (${
+    addContent(${/* slot name */ is.expression} ?? "", async function* generateSlottedContent(${
```

Comment:
Amazingly this does not conflict with other functions in the same scope because `function` names follow `var` semantics so you can just keep redeclaring it. ❤️ JS